ssm/lds.py

```diff
@@ -681,51 +808,75 @@ def _objective(params, itr):
             optimizer(_objective, self.params, num_iters=emission_optimizer_maxiter,
                       state=optimizer_state, full_output=True, step_size=0.01)
 
+    def _laplace_em_elbo(self,
+                         variational_posterior,
+                         datas,
+                         inputs,
+                         masks,
+                         tags,
+                         n_samples=1,
+                         expectations=None):
+
+        def estimate_expected_log_joint(n_samples):
+            exp_log_joint = 0.0
+            for sample in range(n_samples):
+
+                # sample continuous states
+                continuous_samples = variational_posterior.sample_continuous_states()
+                discrete_expectations = variational_posterior.mean_discrete_states
+
+                # log p(theta)
+                exp_log_joint += self.log_prior()
+
+                for x, (Ez, Ezzp1, _), data, input, mask, tag in \
+                    zip(continuous_samples, discrete_expectations, datas, inputs, masks, tags):
+
+                    # The "mask" for x is all ones
+                    x_mask = np.ones_like(x, dtype=bool)
+                    log_pi0 = self.init_state_distn.log_initial_state_distn
+                    log_Ps = self.transitions.log_transition_matrices(x, input, x_mask, tag)
+                    log_likes = self.dynamics.log_likelihoods(x, input, x_mask, tag)
+                    log_likes += self.emissions.log_likelihoods(data, input, mask, tag, x)
+
+                    # Compute the expected log probability
+                    exp_log_joint += np.sum(Ez[0] * log_pi0)
+                    exp_log_joint += np.sum(Ezzp1 * log_Ps)
+                    exp_log_joint += np.sum(Ez * log_likes)
+            return exp_log_joint / n_samples
+
+        def estimate_entropy(n_samples):
```

Comment:
I don't think we need to support the Monte Carlo estimate of the entropy anymore.  Moreover, the expectations can be computed inside the `variational_posterior.entropy()` function, so I would just replace this with,
```
return estimate_expected_log_joint(n_samples) + variational_posterior.entropy()
```